torch/testing/_internal/common_fsdp.py

```diff
@@ -907,6 +907,18 @@ def patch_reduce_scatter(new_reduce_scatter_tensor: Callable):
         dist.reduce_scatter_tensor = orig_reduce_scatter
 
 
+@contextlib.contextmanager
+def patch_all_reduce(new_all_reduce: Callable):
```

Comment:
hopefully after https://github.com/pytorch/pytorch/pull/125475 landed lots of these would be simplified!